src/steps/02-EvaluationCriteria/MRR/CertificationPOCTypeForm.vue

```diff
@@ -0,0 +1,415 @@
+    const savedData = {} as FairOpportunityDTO;
+    const savedFairOpportunity = AcquisitionPackage.fairOpportunity as FairOpportunityDTO;
+    savedData[this.POCTypePropName] = savedFairOpportunity[this.POCTypePropName] || "";
+    return hasChanges(currentData, savedData);
+  }
+
+  private hasCurrentContactFormChanged(): boolean {
+    return hasChanges(this.currentContactFormData, this.certificationPOCContactDTO);
+  }
+
+  /**
+   * Checks to see if the certification POC type changed and takes one of
+   * the actions below.
+   * 1. If the user updates the POC type from "NEW" to one of the other 3 types, then
+   *      DELETES the contact associated with "NEW" and updates the fair opportunity
+   *      properties accordingly
+   * 2. If the user changes the POC type from one of the other 3 types to "NEW",
+   *      then creates a NEW contact and updates the fair opportunity
+   *      properties accordingly
+   * 3. If the user changes the POC type among one of the 3 existing contact types,
+   *      then just updates the fair opportunity properties.
+   * 4. If the POC type has not changed, then checks if any of the form data has changed
+   *      and makes a call to save the changed data.
+   */
+  @Watch('_saveForm')
+  protected async saveOnLeave(): Promise<boolean> {
+    try {
+      await AcquisitionPackage.setValidateNow(true);
+      const savedPOCType =
+          (AcquisitionPackage.fairOpportunity as FairOpportunityDTO)[this.POCTypePropName];
+      const fairOpportunity = {} as FairOpportunityDTO;
+      if (this.selectedOptionType === "NEW") {
+        // user changed from other types to NEW or could be first time filling the form.
+        if (this.hasCurrentContactFormChanged()) {
+          let savedContact = await ContactData.saveContact(this.currentContactFormData);
+          savedContact = convertColumnReferencesToValues(savedContact);
+          fairOpportunity[this.POCTypePropName] = this.selectedOptionType;
+          fairOpportunity[this.POCPropName] = savedContact.sys_id;
+          await AcquisitionPackage.setFairOpportunity(fairOpportunity);
+          this.selectedOption.value = savedContact.sys_id as string;
+        }
+      } else if (savedPOCType === "NEW") {// user has changed from NEW to other type
+        // should not delete the contact because the contact could have been linked to others.
+        fairOpportunity[this.POCTypePropName] = this.selectedOptionType;
+        fairOpportunity[this.POCPropName] = this.selectedOption?.value;
+        await AcquisitionPackage.setFairOpportunity(fairOpportunity);
+      } else { // user has changed across one of the 3 existing contact types (NOT NEW)
+        if (this.hasFairOpportunityDataChanged()) {
+          fairOpportunity[this.POCTypePropName] = this.selectedOptionType;
+          fairOpportunity[this.POCPropName] = this.selectedOption?.value;
+          await AcquisitionPackage.setFairOpportunity(fairOpportunity)
```

Comment:
These 3 lines (254-256) are exactly the same as the block above (249-451), which is almost exactly the same as the first block for checking if it's NEW, which the only difference is it first saves the new contact and then uses the savedContact's sys_id to save with `setFairOpportunity`. Please DRY this up.

Reply:
Since we are not deleting the contacts, I can remove the "else if" block. And will further refactor this.